contracts/DrugItem.sol

```diff
@@ -47,20 +47,37 @@ contract DrugItem is Secondary {
     }
 
     function logHandover(
+        address _from,
         address _to,
         ParticipantCategory _participantCategory
     )
         public
         onlyPrimary
     {
-        require(_participantCategory != ParticipantCategory.Vendor, "Drug item can't be handed over back to any vendor");
+        require(_participantCategory != ParticipantCategory.Vendor, "Drug item can't be handed over to a vendor");
+        require(handoverLog.length == 0 || getLastHandover().to.id == _from, "Handover must be done by the current drug item holder");
+        require(
+            handoverLog.length < 2 ||
+                getLastHandover().to.category != ParticipantCategory.Carrier ||
+                hasTransitConditionsForLastHandover(),
+            "Transit conditions must be logged before next handover"
+        );
         handoverLog.push(Handover(Participant(_to, _participantCategory), now));
```

Comment:
Shouldn't we emit every drug item handover?

Reply:
It's possible but not necessary. We're saving all the data in these data structures because we need to read from it later on. What added value do you see in emitting events here?